utils/treesitter/queries/highlights.scm

```diff
@@ -0,0 +1,112 @@
+; Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+; Exceptions. See /LICENSE for license information.
+; SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+(comment) @comment
+(builtin_type) @type.builtin
+(sized_type_literal) @type.builtin
+(bool_literal) @constant.builtin
+(escape_sequence) @constant.character.escape
+(string_literal) @string
+(integer_literal) @constant.builtin
+(float_literal) @constant.builtin
+
+; function declaration or call expression => function
+(function_declaration (declared_name (ident) @function))
+(call_expression (ident) @function)
+
+; upper case => type
+((ident) @type
+  (#match? @type "^[A-Z]"))
+
+; lower case => variable
+((ident) @variable
+  (#match? @variable "^[a-z_]"))
```

Comment:
The case is not a reliable indicator here. Would be much better to do what you do with functions: identifiers from patterns should be highlighted like variables, identifiers from class/interface/constraint/etc. declarations should be highlighted like types.

Reply:
Good idea. I would like to do it in a follow up.

Reply:
Please include a TODO comment to indicate what is expected to change.